src/pages/settings/Wallet/ExpensifyCardPage.tsx

```diff
@@ -162,7 +162,7 @@ function ExpensifyCardPage({
         <ScreenWrapper testID={ExpensifyCardPage.displayName}>
             <HeaderWithBackButton
                 title={pageTitle}
-                onBackButtonPress={() => Navigation.goBack()}
+                onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_WALLET)}
```

Comment:
Coming from this issue https://github.com/Expensify/App/issues/58260, adding navigation here caused the app to get stuck in loop when accessing the assigned card from the Wallet page. Instead, we should have used `goBack` to navigate to `SETTINGS_WALLET`